File: INF-143/contest_7/F/main.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
	string entrada, num, num1, num2;

	while(getline(cin, entrada, '\n'))
	{

		if(entrada[entrada.size() - 1] != ' ')
			entrada += " ";

		double costo_total = 0, precio_total = 0, ganancia;

		for(int i = 0; i < entrada.size(); i++)
		{
			if(entrada[i] == ' ')
			{
				if(i == 6)
					precio_total += stod(num);
				else if(i == entrada.size() - 1)
					costo_total += stod(num);
				else
				{
					costo_total += stod(num.substr(0, 6));
					precio_total += stod(num.substr(7, num.size()));
				}

				num = "";
			}
			else
				num += entrada[i];
		}
		ganancia = 100 * abs(costo_total - precio_total) / precio_total;
		cout << floor(ganancia) << "\n";
	}

	return 0;
}

File: INF-143/contest_7/F/main.py
import math as m
 
def sol():
    cad = input()
 
    if cad[len(cad) - 1] == " ":
        vin = cad[0:len(cad) - 2].split(" ")
    else:
        vin = cad[0:len(cad) - 1].split(" ")
 
    costo_total = float(vin[len(vin) - 1])
    precio_total = float(vin[0] + "")
 
    for i in range(1, len(vin) - 1):
        a = float(vin[i].split(",")[0])
        b = float(vin[i].split(",")[1])
        costo_total += a
        precio_total += b
    ganancia = abs(costo_total - precio_total)
    ans = m.floor((100 * ganancia) / precio_total)
    print(ans)
 
while True:
    try:
        sol()
    except EOFError:
        break